mods/default/functions.lua

```diff
@@ -213,16 +213,27 @@ function default.register_fence(name, def)
 	-- Allow almost everything to be overridden
 	local default_fields = {
 		paramtype = "light",
-		drawtype = "fencelike",
+		drawtype = "nodebox",
+		node_box = {
+			type = "connected",
+			fixed = {{-1/8, -1/2, -1/8, 1/8, 1/2, 1/8}},
+			-- connect_up =
+			-- connect_down =
+			connect_n = {{-1/16,3/16,1/8,1/16,5/16,1/2},
+				{-1/16,-5/16,1/8,1/16,-3/16,1/2}},
+			connect_e = {{1/8,3/16,-1/16,1/2,5/16,1/16},
+				{1/8,-5/16,-1/16,1/2,-3/16,1/16}},
+			connect_w = {{-1/2,3/16,-1/16,-1/8,5/16,1/16},
+				{-1/2,-5/16,-1/16,-1/8,-3/16,1/16}},
+			connect_s = {{-1/16,3/16,-1/2,1/16,5/16,-1/8},
+				{-1/16,-5/16,-1/2,1/16,-3/16,-1/8}},
+		},
+		connects_to = {"group:fence", "group:wood", "group:tree"},
```

Comment:
walls are made out of stone:#796

https://github.com/sofar/minetest_game/commit/0c7d4418c9fa5fdacafa86490c7fd70b59ef45c4#diff-c967da67b272f4b656ff35c5d13941cdR33

Which is why I proposed to add these basic wall nodes :)